benchmark/generate_diff.php

```diff
@@ -60,6 +61,24 @@ function formatDiff(?int $baseInstructions, int $headInstructions): string {
     return sprintf('%.2f%%', $instructionDiff / $baseInstructions * 100);
 }
 
+function find_benchmarked_commit_hash(string $commitHash): ?string {
+    $repeat = 10;
+
+    while (true) {
+        if ($repeat-- <= 0) {
+            fwrite(STDERR, "Count not find benchmarked commit hash\n");
+            exit(1);
+        }
+        $summaryFile = __DIR__ . '/repos/data/' . substr($commitHash, 0, 2) . '/' . $commitHash . '/summary.json';
+        if (file_exists($summaryFile)) {
+            break;
+        }
+        $commitHash = trim(runCommand(['git', 'rev-parse', $commitHash . '^'], dirname(__DIR__))->stdout);
```

Comment:
will this work with https://github.com/actions/checkout/blob/main/README.md?plain=1#L95?

Reply:
No, but see:

https://github.com/php/php-src/blob/36ae82b73e2ca97fdda2a78d114a578ce01c81f5/.github/workflows/push.yml#L271-L272

Reply:
I see..

Reply:
IMO this PR should target lower bugfix branches too and the `fetch-depth: 0` is present in PHP-8.3 and higher only.

Reply:
I don't mind backporting this to 8.3.